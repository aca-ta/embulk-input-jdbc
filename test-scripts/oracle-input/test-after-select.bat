sqlplus TEST_USER/XXXXXXXX@localhost/TESTDB @test.sql

del data\test000.00.csv
CALL embulk run test-after-select.yml

echo "diff data/test_expected.csv data/test000.00.csv"
diff data/test_expected.csv data/test000.00.csv

IF "%ERRORLEVEL%" == "0" (ECHO "OK!") ELSE (ECHO "embulk-input-oracle (after-select) FAILED!")

sqlplus TEST_USER/XXXXXXXX@localhost/TESTDB @select.sql

echo "diff data/test-after-select_expected.txt data/temp.txt"
diff data/test-after-select_expected.txt data/temp.txt

IF "%ERRORLEVEL%" == "0" ECHO "OK!"

IF "%ERRORLEVEL%" == "0" (ECHO "OK!") ELSE (ECHO "embulk-input-oracle (after-select) FAILED!")
