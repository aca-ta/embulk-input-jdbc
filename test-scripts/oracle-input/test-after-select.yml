in:
  type: oracle
  driver_path: 'driver/ojdbc7.jar'
  host: localhost
  database: TESTDB
  user: TEST_USER
  password: XXXXXXXX
  table: EMBULK_INPUT
  select: "*"
  column_options:
    ID: {type: string}
    NUM: {type: string}
    DT: {type: string, timestamp_format: '%Y-%m-%d', timezone: "+0900"}
    DTTM0: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S', timezone: "+0900"}
    DTTM3: {type: string, timestamp_format: '%Y-%m-%d %H:%M:%S.%3N', timezone: "+0900"}
  after_select: update EMBULK_INPUT set STR = 'xx'
out:
  type: file
  path_prefix: data/test
  file_ext: csv
  formatter:
    type: csv
