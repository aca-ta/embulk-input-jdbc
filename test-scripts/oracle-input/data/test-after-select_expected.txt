
        ID        NUM STR      VARSTR   DT         DTTM0                        
---------- ---------- -------- -------- ---------- -------------------          
DTTM3                                                                           
-----------------------------                                                   
         1      123.4 xx       TEST1    2015-04-24 2015-04-24 01:02:03          
2015-04-24 01:02:03.123                                                         
                                                                                
         2 1234567890 xx       TEST9999 2015-12-31 2015-12-31 23:59:59          
2015-12-31 23:59:59.999                                                         
                                                                                
         3            xx                                                        
                                                                                
                                                                                

